### DISCUSSION TITLE
Joomla 4 loading on online server only minified files / works local though

### DISCUSSION THREAD

**Discussion by olivdee** [joomla/joomla-cms#38851] [General]
I was wondering if this is a bug, or I might be missing something:
Currently experimenting with Joomla 4, and when using a custom edited editor.css file for TinyMCE, this file gets never loaded, instead it always reverts to editor.min.css (which has obviously not been edited). No matter where I place my custom editor.css it does not work. Not inside \media\templates\site\cassiopeia\css\, nor inside a child templates folder \media\templates\site\cassiopeia_child\css\. 

Tried this on 2 different online servers, with and without a child template. 
It only works on my local XAMPP setup. 
I am totally confused... 
Any hint/suggestion is greatly appreciated!

**Comment by olivdee** [comment]
Follow up: 
If I change editor.css to editor.min.css inside \media\templates\site\cassiopeia_child\css\ it works on the live server.
Should the description in Plugins > Editor - TinyMCE rather be "By default the Plugin looks for an editor.css**/editor.min.css** file. If it can't find one in the default template CSS folder, it loads the editor.css**/editor.min.css** file from the system template."? With a short sentence explaining when it is the case editors.min.css gets precedence?

**Comment by Fedik** [comment]
It forcing `min` version (if it exists) when Debug is Off https://github.com/joomla/joomla-cms/blob/f520c98c64efb538e70e918acffb60d07e4d6b1e/plugins/editors/tinymce/src/PluginTraits/ResolveFiles.php#L41

You have 2 option:
- remove `min` version
- create your `min` version

**Reply by olivdee** [reply]
Great, thanks for clarifying this by showing the code. I figured out your suggested options quickly myself by consulting devtools/networktab, but still from a usability point of view, and especially for users not very tech-savvy, the description info of the related field in TinyMCE is (in my opinion) not optimal. I can imagine this driving people crazy, searching for days etc.
What do you think?

**Reply by brianteeman** [reply]
the code is wrong

**Reply by Fedik** [reply]
Yeah, it can be annoying. We probably should not force min version for user custom css for editor and user.css for template. 
Method `resolveFileUrl`  coded in way that force it https://github.com/joomla/joomla-cms/blob/f520c98c64efb538e70e918acffb60d07e4d6b1e/plugins/editors/tinymce/src/PluginTraits/ResolveFiles.php#L81

**Reply by brianteeman** [reply]
thats just plain wrong. The logic should be
Does an editor.min.css exist - use it
Does an editor.css exist - use it
Is debug enabled and does an editor.css exist - use it
Is debug enabled and does an editor.min.css exist - use it
use default system editor.css

**Reply by olivdee** [reply]
if I understand this correct, and if this is the suggested precedence (with "upper/higher the list" = higher precedence), then there should be definitely an adjustment in the description in Plugins > Editor - TinyMCE rather be "By default the Plugin looks for an editor.css file. If it can't find one in the default template CSS folder, it loads the editor.css file from the system template." where this is made a bit clearer?

**Reply by olivdee** [reply]
just to put this ad acta, is my suggestion changing slightly the description to be more clear valid?
and: on another clean install on an online server (J!4.2.3) I tried to use the next option in the plugin "Custom CSS Classes": this does not work either... I am adding a custom.css inside media/templates/site/cassiopeia/css/custom.css - and it does not being loaded. I tried all kinds of combinations, entering plain "custom.css" in the field (as it explains "Enter a file name..."), but tried also the aforementioned path. Also tried both mentioned ways in combination with "Template CSS Classes" ON & OFF - still no luck. Either I am far to stupid/blind to get this working, or all 3 servers (1 xampp, 2 different providers) are buggy/faulty, or there is a bug...? Do I have to clean cache after each combination? Could somebody please confirm if this works, because this is driving my mad...

**Reply by dgrammatiko** [reply]
@Fedik @brianteeman  the code is working as Brian explains what should be the correct logic and the logic is system wide: 

```
Does an editor.min.css exist - use it
Does an editor.css exist - use it
Is debug enabled and does an editor.css exist - use it
Is debug enabled and does an editor.min.css exist - use it
```

The problem is not the code Fedir mantioned but rather some missing functionality in the file manager of the templates. What should be there is that when a user saves either `user.css` or `editor.css` the corresponding `.min.css` should be generated automatically (and probably an alert if a user tries to edit directly the minified file explaining that they should instead edit the non minified file). Theoretically that should be a couple of conditionals in the model template.php

My 2c

**Reply by Fedik** [reply]
@olivdee yeap the description is misleading

> the corresponding .min.css should be generated automatically

@dgrammatiko it may help for backend template editor, but problem that people do not know that there should be `min.css` version, when they edit manualy (when parent template have `min.css`).

**Reply by olivdee** [reply]
This is exactly what I mean - it is great to add the described fix (automatic generation of .min.css), but please add this info also into the description of the corresponding field(s), so that somebody editing manually know about it. Is my last remark about the custom.css related? And: is there already a PR for all this I can follow? Thanks!
